woocommerce/woocommerce-gutenberg-products-block
# Use @wordpress/base-styles and @automattic/color-studio as a base for our styles

Fixes #1331.

This PR:
* Replaces several SCSS variables and hardcoded CSS values with variables from `@wordpress/base-styles` and `@automattic/color-studio` in order to align our colors with Gutenberg and WooCommerce brand. Notice that means this PR introduces some visual changes, but none should be too drastic since I tried to always use the closer match. We used to have many different shades of gray that are now reduced to 7, that's in line with what Gutenberg did to reduce the number of colors (https://github.com/WordPress/gutenberg/pull/23648).
* Decreases the specificity of price slider styles, so they are easier to theme + adds [documentation](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/update/styles-source/docs/theming/all-products-and-filters.md) on how to achieve it.
* Adds a SCSS mixin (`encode-color()`) so SCSS variables can be used in SVG data-urls.

Tip: it might be easier to review this PR with the `Hide whitespace changes` option on GH.

#### Accessibility

I tested color contrast in several blocks to ensure there were no regressions.

### Screenshots
Some screenshots:
![imatge](https://user-images.githubusercontent.com/3616980/96609382-c15aa800-12fa-11eb-8019-b3bb2586b59a.png)
![imatge](https://user-images.githubusercontent.com/3616980/96609306-ad16ab00-12fa-11eb-9954-2bf4d05be64a.png)
![imatge](https://user-images.githubusercontent.com/3616980/96609184-8789a180-12fa-11eb-84a7-6028273186f0.png)
![imatge](https://user-images.githubusercontent.com/3616980/96609218-91130980-12fa-11eb-8595-72b845ed7337.png)



### How to test the changes in this Pull Request:

Test all blocks in editor/frontend and verify there are no evident issues with its display (layout completely broken or colors that are not correctly assigned). You can copy & paste [this gist](https://gist.github.com/Aljullu/d9c76ed510ae6973bd41dcc4184e8eb2) into the editor to automatically add all blocks. You might need to paste it with <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>V</kbd> so styles are removed and the editor can transform it into blocks.

### Changelog

> Updated several colors in blocks to align them with Gutenberg colors and WooCommerce brand.
